src/fdosecrets/objects/Collection.cpp

```diff
@@ -602,9 +625,10 @@ namespace FdoSecrets
 
     void Collection::cleanupConnections()
     {
-        if (m_exposedGroup) {
-            m_exposedGroup->disconnect(this);
+        while (!m_connections.isEmpty()) {
+            disconnect(m_connections.takeFirst());
         }
```

Comment:
Technically you can just call `disconnect()` to remove all connections to/from this QObject. It's rather robust and avoids having to track all the connections.

Reply:
In this case, only some of the incoming signals (from `Database`, `Metadata`, `Group`, but not `DatabaseWidget`) need to be disconnected, so I can't use the catch-all `disconnect`.

Without `m_connections`, it will be multiple `otherobject->disconnect(this)` calls with a recursive helper function to disconnect from groups. I find it easy to mismatch what's connected and what's disconnected.

But it's also easy to forget if tracking connections manually. :/ So I don't have a strong opinion on this.

Anyway, since you asked, I'll change to the multiple `otherobject->disconnect(this)` calls version, which at least feels more natural to me.

Reply:
Just found out `groupsResurcive`. I'll use that.